components/datetime/src/provider/calendar/mod.rs

```diff
@@ -15,31 +15,29 @@ pub use symbols::*;
 
 #[icu_provider::data_struct(DatePatternsV1Marker = "datetime/lengths@1")]
 #[derive(Debug, PartialEq, Clone, Default)]
-#[cfg_attr(
-    feature = "provider_serde",
-    derive(serde::Serialize, serde::Deserialize)
-)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serialize", derive(serde::Deserialize))]
 pub struct DatePatternsV1<'data> {
```

Comment:
Thought/Suggestion: Is it time to move these into `#[icu_provider::data_struct]`?

Reply:
...maybe. the macro could probably even automatically mutate serde(borrow) things. I worry about it making the code more opaque though

and we still need these derives on non-data-struct subfields of data structs.